pkg/deployment/config.go

```diff
@@ -175,6 +229,23 @@ func NewConfigFromMap(configMap map[string]string) (*Config, error) {
 			return nil, fmt.Errorf("unsupported %s value: %q", defaultAffinityTypeKey, affinity)
 		}
 	}
+	if err := yaml.Unmarshal([]byte(runtimeClassNames), &nc.RuntimeClassNames); err != nil {
+		return nil, fmt.Errorf("%v cannot be parsed, please check the format", RuntimeClassNameKey)
+	}
+	if len(nc.RuntimeClassNames) > 0 {
+		for class, rcn := range nc.RuntimeClassNames {
+			if len(rcn.Selector) == 0 {
+				continue
+			}
+			if slices.Contains(apimachineryvalidation.NameIsDNSSubdomain(class, false), class) {
+				return nil, fmt.Errorf("%v %v selector not valid DNSSubdomain", RuntimeClassNameKey, class)
+			}
```

Comment:
This is confusing and it doesn't seem right.

Looking at the code _any_ string returned by `apimachineryvalidation.NameIsDNSSubdomain` implies there's an error.

https://github.com/kubernetes/apimachinery/blob/37988e577e160bf7749437141fee1d6b8ca15dd6/pkg/util/validation/validation.go#L215-L223

Reply:
I'm having some weird results with the function.

```go
package main

import (
        "fmt"

        apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
)

func main() {
        for _, v := range []string{"a", "gvisor", "kata"} {
                fmt.Println(v, apimachineryvalidation.NameIsDNSSubdomain(v, false))
        }
}
```

results in 

```
🐚(2) go run .
a []
gvisor []
kata []
```

looking at the tests, `a` should succeed
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation_test.go#L55

I set it how it is currently around the tests, as the tests seems to match up right.
Happy to update and understand what's happening here!

Reply:
empty array means it's valid - so I think we should just check for an non-empty array

Reply:
OK, revisited it again and according to the tests the behaviour should be in place now.
Updated in 6b6ed607b